### DISCUSSION TITLE
How can I disable a lsp server if a file does not exist?

### DISCUSSION THREAD

**Discussion by mapringg** [LazyVim/LazyVim#409] [Q&A] [answered]
I would like to disable the stylelint server when there is no stylelint config. I got it working with local neoconf, but is there a way of doing that from my plugin file:

```
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      stylelint_lsp = {
        filetypes = { "css", "less", "scss" },
      },
    },
  },
}
```

Edit: I tried adding root_dir, but that did not work:
```
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      stylelint_lsp = {
        root_dir = require("lspconfig.util").root_pattern(".stylelintrc"),
        filetypes = { "css", "less", "scss" },
      },
    },
  },
}
````

**Comment by folke** [comment, marked as answer]
You'll need to disable neoconf to make that root_dir work

**Reply by mapringg** [reply]
Thank you 🍻

**Reply by dpetka2001** [reply]
@folke Just for clarification because I also might be interested in this, do you mean disable the neoconf plugin as a whole, or just remove the project .neoconf file from the root_dir?

**Reply by folke** [reply]
remove the whole plugin

**Reply by danielo515** [reply]
That's too bad, because I can see myself wanting both. Basically because neoconf is very convenient, but sometimes you just need more flexibility than a JSON file
